gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/RecommendedProductDependenciesPlugin.java

```diff
@@ -30,18 +33,32 @@ public final void apply(Project project) {
                 .create("recommendedProductDependencies", RecommendedProductDependenciesExtension.class, project);
 
         project.getPluginManager().withPlugin("java", _plugin -> {
-            TaskProvider<ConfigureProductDependenciesTask> configureProductDependenciesTask = project.getTasks()
-                    .register("configureProductDependencies", ConfigureProductDependenciesTask.class, cmt -> {
-                        cmt.setProductDependencies(ext.getRecommendedProductDependenciesProvider());
-                    });
+            Provider<Directory> dir = project.getLayout().getBuildDirectory().dir("product-dependencies");
+            TaskProvider<CompileRecommendedProductDependencies> compilePdeps = project.getTasks()
+                    .register(
+                            "compileRecommendedProductDependencies",
+                            CompileRecommendedProductDependencies.class,
+                            task -> {
+                                task.getRecommendedProductDependencies()
+                                        .set(ext.getRecommendedProductDependenciesProvider());
+                                task.getOutputFile()
+                                        .set(dir.map(directory -> directory.file(
+                                                RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY
+                                                        + "/product-dependencies.json")));
+                            });
 
-            // Ensure that the jar task depends on this wiring task
             project.getTasks()
-                    .withType(Jar.class)
-                    .named(JavaPlugin.JAR_TASK_NAME)
-                    .configure(jar -> {
-                        jar.dependsOn(configureProductDependenciesTask);
-                    });
+                    .named(
```

Comment:
If we remove writing things to the jar manifest, won't we break any products that are consuming these jars that have not updated to this version of sls-packaging?